### PR TITLE
Global configuration file parsing

### PR DESCRIPTION
**Pull request by Jim3Things**

### CODE REVIEW

File: internal/config/settings.go [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+        "ReadGlobalConfig")
+    defer span.End()
+
+    if err := viper.ReadInConfig(); err != nil {
+        if _, ok := err.(viper.ConfigFileNotFoundError); ok {
+            // Config file not found; we'll just use the default values
+            span.AddEvent(
+                ctx,
+                fmt.Sprintf(
+                    "No config file found at %s/%s (%s), applying defaults.\n",
```
Comment (MikeCJohnsonUk) [inline internal/config/settings.go]: Probably minor, but should trace message strings themselves have the "\n"?

File: tools/fetch_imports.cmd [hunk 2]
```diff
@@ -34,4 +34,6 @@ go get google.golang.org/grpc
 
 go get golang.org/x/crypto/...
 
+go get github.com/spf13/viper
```
Comment (MikeCJohnsonUk) [inline tools/fetch_imports.cmd]: Are we at the point we can remove fetch_imports.cmd in favour of fetchall.cmd? That was certainly my intent when adding fetchall.cmd

File: internal/config/settings.go [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+        // Fill in the global configuration object from the configuration file
+        if err = viper.UnmarshalExact(cfg); err != nil {
+            err = fmt.Errorf("unable to decode into struct, %v", err)
+            span.AddEvent(ctx, err.Error())
+            return nil, err
+        }
+    }
+
+    span.AddEvent(ctx,
+        fmt.Sprintf("Configuration Read: \n%v", ToString(cfg)))
```
Comment (MikeCJohnsonUk) [inline internal/config/settings.go]: \n before the %v?

File: internal/config/settings.go (hunk 1 above)
Comment (Jim3Things) [reply internal/config/settings.go]: Good catch.  Will fix.

File: tools/fetch_imports.cmd (hunk 2 above)
Comment (Jim3Things) [reply tools/fetch_imports.cmd]: I think so.  Fwiw, I've been using updateall exclusively.

File: internal/config/settings.go (hunk 3 above)
Comment (Jim3Things) [reply internal/config/settings.go]: That's because the ToString produces a formatted output, the \n ensures that the first line formatting is correct.

**Comment by Jim3Things** [comment]
We can add a command line qualifier ('-showconfig') that would write out the effective config to stdout by using the ToString function.  I'll add that in the next update.

**Comment by Jim3Things** [comment]
I've updated with the CR comments.

File: internal/config/settings.go [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+    DefaultGlobalConfigFile = "cloudchamber.yaml"
+    DefaultConfigType = "yaml"
+    ControllerDefaultPort = 8081
+    InventoryDefaultPort = 8082
+    SimSupportDefaultPort = 8083
+    WebServerDefaultPort = 8084
+    WebServerFEDefaultPort = 8080
+    DefaultHost = ""
+    DefaultRootFilePath = "."
+    DefaultSystemAccount = "Admin"
```
Comment (MikeCJohnsonUk) [inline internal/config/settings.go]: Implies user/account names are not case-sensitive?

Comment (Jim3Things) [reply internal/config/settings.go]: No.  I don't think this component cares.  I've set up some defaults that are supplied if there's no config here, and the .yaml also has values provided.  Some of them are different specifically so I could verify that the .yaml was processed and that I wasn't looking at the result of the default pass.  I figured some of these would change as we get, e.g. the full user support in place.

**Pull request merged**